src/lib/yacc.rs

```diff
@@ -216,18 +239,23 @@ impl YaccParser {
                 }
 
             },
-            None => Err(YaccError{kind: YaccErrorKind::IllegalString, off: i})
+            None => {
+                Err(self.mk_error(YaccErrorKind::IllegalString, i))
+            }
         }
     }
 
-    fn parse_ws(&self, i: usize) -> YaccResult<usize> {
-        match RE_WHITESPACE.find(&self.src[i..]) {
-            Some((s, e)) => {
-                assert!(s == 0);
-                Ok(i + e)
+    fn parse_ws(&mut self, i: usize) -> YaccResult<usize> {
+        let mut j = i;
+        for c in self.src.chars().skip(i) {
```

Comment:
Personally I'm not a fan of the self managed counter, what do you think about this:

```rust
for (j, c) in self.src.chars().enumerate().skip(i) {
    match c {
        ' '  | '\t' => (),
        '\n' | '\r' => self.newlines.push(i),
        _           => return Ok(j)
    }
}
Ok(i)
```